File: Session_04_Repetition/repetition.c
#include <stdio.h>

int main() {
    // 1. For Loop (Fixed count execution)
    printf("--- For Loop (1 to 5) ---\n");
    for (int i = 1; i <= 5; i++) {
        printf("%d ", i);
    }
    printf("\n");

    // 2. While Loop (Conditional count execution)
    printf("\n--- While Loop (Countdown) ---\n");
    int count = 3;
    while (count > 0) {
        printf("%d... ", count);
        count--;
    }
    printf("Launch!\n");

    // 3. Do-While Loop (Validating user input)
    printf("\n--- Do-While Loop validation input ---\n");
    int user_input;
    do {
        printf("Choose Option (1 or 2): ");
        user_input = 2; // Simulate valid user input
        printf("%d\n", user_input);
    } while (user_input != 1 && user_input != 2);
    
    printf("Valid option selected!\n");

    // 4. Nested Loops: Star shape pattern (Popular Binus exam question!)
    printf("\n--- Nested Loops: Triangle Pattern ---\n");
    int size = 5;
    for (int row = 1; row <= size; row++) {
        for (int col = 1; col <= row; col++) {
            printf("*");
        }
        printf("\n"); // Move to next line
    }

    return 0;
}

File: Session_04_Repetition/repetition.py
# 1. For Loop (Python uses range() generator)
print("--- For Loop (1 to 5) ---")
for i in range(1, 6): # Starts at 1, ends before 6
    print(f"{i} ", end="")
print()

# 2. While Loop
print("\n--- While Loop (Countdown) ---")
count = 3
while count > 0:
    print(f"{count}... ", end="")
    count -= 1
print("Launch!")

# 3. Python does not have do-while loops.
# To replicate do-while, we use "while True" with a conditional "break"
print("\n--- Simulating Do-While in Python ---")
while True:
    print("Choose Option (1 or 2): ", end="")
    user_input = 2 # Simulate valid input
    print(user_input)
    if user_input == 1 or user_input == 2:
        break # Exit loop
print("Valid option selected!")

# 4. Nested Loops: Star shape pattern
print("\n--- Nested Loops: Triangle Pattern ---")
size = 5
for row in range(1, size + 1):
    for col in range(1, row + 1):
        print("*", end="")
    print()
